src/models/CompanyProfile/CompanyProfileRepository.ts

```diff
@@ -50,7 +63,11 @@ export const CompanyProfileRepository = {
     }
   },
   findById: async (id: number) => {
-    return CompanyProfile.findOne({ where: { id: id } });
+    const companyProfile: CompanyProfile | null = await CompanyProfile.findOne(
```

Comment:
haría que estos metodos retornen solo un tipo, no 2, y que el not found sea un tipo de excepcion